src/ngraph/op/fused/lstm_sequence.cpp

```diff
@@ -0,0 +1,230 @@
+                                                               m_hidden_size,
+                                                               B,
+                                                               P,
+                                                               m_activations,
+                                                               m_activations_alpha,
+                                                               m_activations_beta,
+                                                               m_clip_threshold,
+                                                               m_input_forget);
+
+        shared_ptr<Node> H = get_output_element(lstm_cell, 0);
+        shared_ptr<Node> C = get_output_element(lstm_cell, 1);
+
+        // Expand tensors with empty outermost dim, so we can later concatenate
+        // them.
+        // Mask hidden state tensor in order to handle mixed sequence lengths.
+        // This results in zeroing out values in batches with sequence shorter
+        // than current time_step.
+        h_list.push_back(get_masked_node(builder::expand_dims(H), time_step, 1));
+        // Reference implementation in ONNX Runtime doesn't mask values of Y_h
+        // and Y_c outputs, thus here we make sure that only appropriate batches
+        // (in respect to its sequence length) are updated. Those batches which
+        // has shorter sequences preserve the last value.
+        H_t = get_masked_node(H, time_step, 0, H_t);
+        C_t = get_masked_node(C, time_step, 0, C_t);
+        time_step++;
+    }
+    // The tensor that concats all the intermediate output values of the hidden.
+    // It has shape [seq_length, batch_size, hidden_size]
+    shared_ptr<Node> Y{make_shared<op::Concat>(h_list, 0)};
+
+    // Get back the original order of the output data.
+    if (is_reverse)
+    {
+        Y = make_shared<op::ReverseSequence>(Y, seq_lengths, 1 /*batch_axis*/, 0 /*seq_axis*/);
+    }
+
+    // Expand Y so that it has expected shape:
+    // [seq_length, num_directions, batch_size, hidden_size]
+    Y = builder::expand_dims(Y, 1);
+
+    // expand H_t and C_t so that it has expected shape:
+    // [num_directions, batch_size, hidden_size]
+    auto Y_h = builder::expand_dims(H_t);
+    auto Y_c = builder::expand_dims(C_t);
+    return {Y, Y_h, Y_c};
+}
+
+shared_ptr<Node> op::LSTMSequence::prepare_input(Output<Node> node, bool is_reverse) const
+{
+    // In bidirectional mode inputs are stacked together, so we must split them.
+    shared_ptr<Node> tmp = node.get_node_shared_ptr();
+    if (m_direction == "bidirectional")
+    {
+        tmp = builder::split(node, 2).at(is_reverse ? 1 : 0);
+    }
+    // Since we have forward LSTM we can squeeze `num_directions` axis from inputs.
+    return builder::squeeze(tmp);
+}
+
+void op::LSTMSequence::validate_and_infer_types()
+{
+    NGRAPH_CHECK(m_direction == "bidirectional" || m_direction == "forward" ||
```

Comment:
Replace with constants if these can't be enums.

Reply:
m_direction is a enum now.